dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -34,6 +35,10 @@ inline std::string IndexFileErrMsg(const std::string& index_path, int64_t line,
 namespace detail {
 namespace wds {
 
+bool VerifyIndexVersionString(const std::string& index_string) {
+  return std::regex_match(index_string, std::regex(R"(v\d{1,2}\.\d{1,2})"));
```

Comment:
we had binary compatibility issues with using regex at some point. It was in the numpy reader, I can't recall what was the problem

Reply:
The problem is that it simply crashes when we compile DALI with old C++ ABI, which is forced upon us by manylinux.

Reply:
If you use the code @mzient proposed for parsing this additional check can probably go away.